web/src/stream_list_sort.ts

```diff
@@ -218,38 +261,70 @@ export function sort_groups(stream_ids: number[], search_term: string): StreamLi
     };
 }
 
-function maybe_get_stream_id(i: number): number | undefined {
-    if (i < 0 || i >= all_streams.length) {
-        return undefined;
-    }
-
-    return all_streams[i];
-}
-
-export function first_stream_id(): number | undefined {
-    return maybe_get_stream_id(0);
+export function first_row(): StreamListRow | undefined {
+    return all_rows.at(0);
 }
 
-export function prev_stream_id(stream_id: number): number | undefined {
-    const i = all_streams.indexOf(stream_id);
-
-    if (i === -1) {
-        return undefined;
+export function prev_row(
+    row: StreamListRow,
+    sections_showing_inactive: Set<string>,
+    collapsed_sections: Set<string>,
+): StreamListRow | undefined {
+    let i = all_rows.indexOf(row);
+    const section_id_map = stream_id_to_section_id();
+    while (i > 0) {
+        i -= 1;
+        const prev_row = all_rows[i]!;
+        if (prev_row.type === "stream") {
+            const stream_id = prev_row.stream_id;
+            assert(stream_id !== undefined);
+            const section = section_id_map.get(stream_id)!;
+            if (collapsed_sections.has(section.id)) {
```

Comment:
So... this is currently skipping channels that we keep shown because they're active when a section is collapsed. Not sure what to do about that yet.

Reply:
Added a commit for that.